### PR TITLE
ENH: Add set_data to EncodedStreamObject

### LINKED ISSUE #656: PyPDF2.utils.PdfReadError: Creating EncodedStreamObject is not currently supported
My task is  to find and replace the text in pdf, I used pyPDF2 package to replace the text, but when I try to replace I'm receiving an error like 

```
Traceback (most recent call last):
  File "c:\practice_python\sample.py", line 41, in <module>
    page.getContents().setData(replaced_text)
  File "C:\Users\Win\AppData\Local\Programs\Python\Python310\lib\site-packages\PyPDF2\generic.py", line 852, in setData
    raise utils.PdfReadError("Creating EncodedStreamObject is not currently supported")
PyPDF2.utils.PdfReadError: Creating EncodedStreamObject is not currently supported
```

and my code is 

```python
from PyPDF2 import PdfFileReader, PdfFileWriter

replacements = [
    ("HARIHARAN S", "<your name>")
]

pdf = PdfFileReader(open("samplenda.pdf", "rb"))
writer = PdfFileWriter() 

for page in pdf.pages:
    contents = page.getContents().getData()
    print(type(contents))
    for (a,b) in replacements:
        replaced_text = contents.replace(bytes(a,'utf-8'), bytes(b,'utf-8'))  # .encode('utf-8')
    print(type(replaced_text))
    page.getContents().setData(replaced_text)
    writer.addPage(page)
    
with open("modified.pdf", "wb") as f:
     writer.write(f)
```

I tried lots of way many times, please help me to solve this error

### PR DESCRIPTION
**Pull request by pubpub-zz**
add set_data() for encoded streams
also, complete  FlateEncode to get all requierd attributes Ease data manipulation without going through ContentStream (slow) closes #656

### CODE REVIEW

**Comment by pubpub-zz** [comment]
@MartinThoma 
the codecov did not worked. Can you rerun the action ?

**Comment by MartinThoma** [comment]
Sorry for the delay  :see_no_evil: 

It looks good to me, but doesn't have full coverage. Should I merge it?

**Comment by pubpub-zz** [comment]
you can for me

File: tests/test_generic.py [hunk 1]
```diff
@@ -1169,3 +1169,33 @@ def test_destination_withoutzoom():
     name = "2021_book_security.pdf"
     reader = PdfReader(BytesIO(get_pdf_from_url(url, name=name)))
     reader.outline
+
+
+def test_encodedstream_set_data():
+    """
+    test the encoded stream set_data to extend data stream
+    checks also the flate_encode
+    """
```
Comment (MartinThoma) [inline tests/test_generic.py:1178]:
```suggestion
    """
    EncodedStreamObject.set_data to extend data stream works.

    Checks also the flate_encode.
    """
```

**Comment by MartinThoma** [comment]
Very nice work! Thank you :hugs:  A lot of people were/are looking forward to this :-)

**Pull request merged**